site/src/pages/docs/advanced-spring-boot-integration.mdx

```diff
@@ -0,0 +1,420 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class ArmeriaConfiguration {
+   public static Connector getConnector(ServletWebServerApplicationContext applicationContext) {
+       final TomcatWebServer container = (TomcatWebServer) applicationContext.getWebServer();
+       container.start();
+       return container.getTomcat().getConnector();
+   }
+
+   @Bean
+   public TomcatService tomcatService(ServletWebServerApplicationContext applicationContext) {
+       return TomcatService.of(getConnector(applicationContext));
+   }
+
+   @Bean
+   public ArmeriaServerConfigurator armeriaServerConfigurator(TomcatService tomcatService) {
+       return sb -> sb.serviceUnder("/", tomcatService);
+   }
+}
+```
+
+Add the following properties to your `application.yml` (or `application.properties`) file.
+This configuration will prevent the exposure of the embedded Tomcat service and instead expose your Armeria service.
+You can explore additional configuration options in <type://ArmeriaSettings>.
+
+```yaml filename=application.yml
+# Prevent the embedded Tomcat from opening a TCP/IP port.
+server:
+  port: -1
+
+armeria:
+  ports:
+    - port: 8080
+      protocols:
+        - HTTP
+```
+
+With this simple configuration, everything is set up. Now you can serve your legacy Spring service through Armeria.
+Although it may not seem like much has changed compared to before, your application is now ready to serve both RPC services
+(such as gRPC or Thrift) and RESTful services via Armeria.
+
+
+## Integrating with Spring Boot
+
+<Tip>
+
+  Visit [armeria-examples/spring-boot-minimal](https://github.com/line/armeria/tree/main/examples/spring-boot-minimal) to
+  find a fully working example.
+
+</Tip>
+
+Even if you don't utilize Spring MVC based on web servlets like Tomcat, there is still significant value in combining Spring and Armeria.
+You can leverage the automatic configuration and dependency injection features of Spring Boot to develop an Armeria application.
+
+First, you need the `armeria-spring-boot3-starter` dependency:
+
+<RequiredDependencies
+  boms={[
+    { groupId: 'com.linecorp.armeria', artifactId: 'armeria-bom' }
+  ]}
+  dependencies={[
+    {
+      groupId: 'com.linecorp.armeria',
+      artifactId: 'armeria-spring-boot3-starter',
+    }
+  ]}
+/>
+
+Armeria's annotated service is similar to a Spring controller.
+You can declare the annotated service as a bean and inject it into the <type://ServerBuilder>.
+Here's an example:
+
+```java filename=TodoAnnotatedService.java
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.annotation.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import java.util.List;
+
+// Make this annotated service a bean.
+@Controller
```

Comment:
Technically we don't need this annotation for an annotated service. How about removing it?
```suggestion
```

Reply:
Thank you for confirming. In the conventional Armeria server setup, developers manually create an `AnnotatedService` to run the server. However, in this code example, an `AnnotatedService` instance is obtained using Spring's dependency injection.

In the following code segment, an instance of `TodoAnnotatedService` is injected as a parameter.

```java filename=ArmeriaConfiguration.java
@Bean
public ArmeriaServerConfigurator armeriaServerConfigurator(TodoAnnotatedService todoAnnotatedService) {
   return serverBuilder -> {
       serverBuilder.serviceUnder("/docs", new DocService());
                    .decorator(LoggingService.newDecorator());
                    .annotatedService("/api", todoAnnotatedService);
   };
}
```

Therefore, it's a Spring-managed component, indicated by the addition of the `@Controller` annotation.

Reply:
Ah, you already left a comment on that. Sorry, I didn't notice it. 😓
But, I'm still worried that this might confuse the user even though we don't need `@Contoller` at all.
What do you think of providing the instance using `@Bean`, then?

Reply:
Thank! It seems like a much better idea. I will make the changes accordingly. However, the `org.springframework.stereotype.Bean` annotation cannot be applied to classes (types), so I applied the `org.springframework.stereotype.Component` annotation instead. Is that okay?

Reply:
(Sorry, I misunderstood you mentioned, so I deleted the previous reply I had posted.)

Thank you for checking! Is there a reason to avoid using `@Controller` or `@Component` for injection? When using `@Bean` for dependency injection, we need to add a method like the following:

```java
@Bean
public TodoAnnotatedService todoAnnotatedService(TodoService todoService) {
    return new TodoAnnotatedService(todoService);
}
```

In this case, `TodoAnnotatedService` needs to receive the dependency `TodoService` as a parameter. This is automatically injected using `@Autowired` when using the `@Component` annotation. Therefore, using `@Bean` might lead to longer code.

I prefered to use `@Controller` or `@Component` to demonstrate that using Spring Web MVC with Armeria makes dependency management convenient.

Reply:
I was expecting to use `@Bean` with a method:
```java
@Bean
Public TodoService todoService() {...}
```
But, it sounds like using `@Component` is much better. 👍